experiments/kube-aws/az-down/experiment/az-down.go

```diff
@@ -0,0 +1,334 @@
+	}
+	azDetails.Subnets = subnetDetails
+
+	// create and assign dummy acl to each subnet vpc
+	for i, subnet := range azDetails.Subnets {
+		output, err := ec2Svc.CreateNetworkAcl(&ec2.CreateNetworkAclInput{VpcId: &subnet.VpcId})
+		if err != nil {
+			log.Errorf("Failed to assign dummy ACM to vpc of az being targeted, err: %v", err)
+			failStep := "Failed to assign dummy ACL to vpc of az being targeted"
+			result.RecordAfterFailure(&chaosDetails, &resultDetails, failStep, clients, &eventsDetails)
+			return
+		}
+		subnet.DummyAclId = fmt.Sprint(output.NetworkAcl.NetworkAclId)
+		azDetails.Subnets[i] = subnet
+		log.Info(fmt.Sprintf("New ACL created and assigned to vpc id: %s.\n%s", subnet.VpcId, output))
+	}
+
+	// get NetworkAclAssociationId and NetworkAclId of subnets in az
+	for _, subnet := range azDetails.Subnets {
+		networkAclDetails, err := getNetworkAclDetails(&ec2Svc, subnet.Id)
+		if err != nil {
+			log.Errorf("Failed to get NetworkAclAssociationId for subnet %s, err: %v", subnet.Id, err)
+			failStep := fmt.Sprintf("Failed to get NetworkAclAssociationId for subnet %s in az being targeted", subnet.Id)
+			result.RecordAfterFailure(&chaosDetails, &resultDetails, failStep, clients, &eventsDetails)
+			return
+		}
+		subnet.NetworkAclAssociationId = networkAclDetails.NetworkAclAssociationId
+		subnet.NetworkAclId = networkAclDetails.NetworkAclId
+	}
+
+	// replace old acl with new acl
+	for _, subnet := range azDetails.Subnets {
+		_, err := ec2Svc.ReplaceNetworkAclAssociation(&ec2.ReplaceNetworkAclAssociationInput{
+			AssociationId: &subnet.Id,
+			NetworkAclId: &subnet.DummyAclId,
+		})
+		if err != nil {
+			log.Errorf("Failed to replace subnet %s ACL with dummy ACL %s, err: %v", subnet.Id, subnet.DummyAclId, err)
+			failStep := fmt.Sprintf("Failed to replace subnet %s ACL with dummy ACL %s in az being targeted", subnet.Id, subnet.DummyAclId)
+			result.RecordAfterFailure(&chaosDetails, &resultDetails, failStep, clients, &eventsDetails)
+			return
+		}
+	}
+
+	// Including the litmus lib
+	if experimentsDetails.ChaosLib == "litmus" {
+		err = litmusLIB.AZDown()
+		if err != nil {
+			log.Errorf("Chaos injection failed, err: %v", err)
+			failStep := "failed in chaos injection phase"
+			result.RecordAfterFailure(&chaosDetails, &resultDetails, failStep, clients, &eventsDetails)
+			return
+		}
+		log.Info("[Confirmation]: chaos has been injected successfully")
+		resultDetails.Verdict = "Pass"
+	} else {
+		log.Error("[Invalid]: Please Provide the correct LIB")
+		failStep := "no match found for specified lib"
+		result.RecordAfterFailure(&chaosDetails, &resultDetails, failStep, clients, &eventsDetails)
+		return
+	}
+
+	//POST-CHAOS APPLICATION STATUS CHECK
+	log.Info("[Status]: Verify that the AUT (Application Under Test) is running (post-chaos)")
+	err = status.CheckApplicationStatus(experimentsDetails.AppNS, experimentsDetails.AppLabel, experimentsDetails.Timeout, experimentsDetails.Delay, clients)
```

Comment:
Should the checkApllicationStatus() made take an additional argument like: status and return the status and error? Might be useful to log and analyse the current status of system against the expected status.